### DISCUSSION TITLE
Single Sign On or Token Login

### DISCUSSION THREAD

**Discussion by Queuecumber** [simple-login/app#1386] [Self-hosting question]
I use oauth2 proxy to allow me to easily log into all the sites I host using google (unless the app natively supports google oauth). I see that google oauth was removed from simple-login. I can use my oauth2 proxy but it currently requires me to log in twice. Is there a way to support single sign on or to take a bearer token of some sort from the proxy and skip the login screen?

**Comment by Queuecumber** [comment]
I think this situation is easily supported by respecting `X-Forwarded-Email` on requests

**Reply by muhlba91** [reply]
@Queuecumber would you mind explaining your implementation here?
i'm looking for a similar solution and would like to know how you solved it. also, does the mobile app work for you when using your oauth2 proxy or is this going to be broken then?

**Reply by Queuecumber** [reply]
Sure, so I think you already found the PR but the gist of it is that a reverse proxy handles the auth and then sets a header on all requests. If simplelogin sees that header it assumes that the auth happened externally and skips it's normal auth process.

It may break non-website usage like the app and browser plugin. I didn't try the app but the browser plugin worked fine using an API token, if the app can do that it would also work fine. You need to tell your authenticating proxy to skip auth for calls to the API endpoint though.

I'm planning to add this feature to anonaddy, where they have said they would merge it, if that helps you. I haven't had time to get beyond the proof of concept phase though
